src/SFML/Window/Win32/JoystickImpl.cpp

```diff
@@ -45,1114 +45,717 @@
+    auto rawInputRetCode = RegisterRawInputDevices(rids, 3, sizeof(RAWINPUTDEVICE));
+    if (rawInputRetCode == 0x0)
     {
-        err() << "Failed to get DirectInput device data: " << result << std::endl;
-
-        return m_state;
+        // :(
+        auto errorCode = GetLastError();
+        err() << "RegisterRawInputDevices returned 0, GetLastError: [" << errorCode
+              << "] Win32 Joystick will not function.";
+        return errorCode;
     }
 
-    // Iterate through all buffered events
-    for (DWORD i = 0; i < eventCount; ++i)
-    {
-        bool eventHandled = false;
-
-        // Get the current state of each axis
-        for (unsigned int j = 0; j < Joystick::AxisCount; ++j)
-        {
-            const auto axis = static_cast<Joystick::Axis>(j);
-            if (m_axes[axis] == static_cast<int>(events[i].dwOfs))
-            {
-                if ((axis == Joystick::Axis::PovX) || (axis == Joystick::Axis::PovY))
-                {
-                    const unsigned short value = LOWORD(events[i].dwData);
-
-                    if (value != 0xFFFF)
-                    {
-                        const float angle = (static_cast<float>(value)) * 3.141592654f / DI_DEGREES / 180.f;
-
-                        m_state.axes[Joystick::Axis::PovX] = std::sin(angle) * 100.f;
-                        m_state.axes[Joystick::Axis::PovY] = std::cos(angle) * 100.f;
-                    }
-                    else
-                    {
-                        m_state.axes[Joystick::Axis::PovX] = 0.f;
-                        m_state.axes[Joystick::Axis::PovY] = 0.f;
-                    }
-                }
-                else
-                {
-                    m_state.axes[axis] = (static_cast<float>(static_cast<short>(events[i].dwData)) + 0.5f) * 100.f / 32767.5f;
-                }
-
-                eventHandled = true;
+    m_sTimerHandle = SetTimer(m_sJoystickHwnd, NULL, 8, nullptr);
 
-                break;
-            }
-        }
+    MSG msg;
 
-        if (eventHandled)
-            continue;
-
-        // Get the current state of each button
-        for (unsigned int j = 0; j < Joystick::ButtonCount; ++j)
-        {
-            if (m_buttons[j] == static_cast<int>(events[i].dwOfs))
-                m_state.buttons[j] = (events[i].dwData != 0);
-        }
+    while (GetMessageW(&msg, m_sJoystickHwnd, 0, 0) != 0x0)
+    {
+        TranslateMessage(&msg);
+        DispatchMessageW(&msg);
     }
 
-    m_state.connected = true;
-
-    return m_state;
+    // Cleanup code
+    KillTimer(m_sJoystickHwnd, NULL);
+    return S_OK;
 }
 
-
 ////////////////////////////////////////////////////////////
-JoystickState JoystickImpl::updateDInputPolled()
+void JoystickImpl::initialize()
 {
-    JoystickState state;
-
-    if (m_device)
+    // we'll be churning through data routinely, so it's simply more efficient to allocate a chunk once and reuse it rather than mallocing over and over again.
```

Comment:
I'm going to have to revisit this or see what the stance of some other folks is on this. I am certain we didn't need `malloc` in this context. Others will come along with better suggestions here so I'll just leave this as a note for an area to dig into.